core/src/kvs/surrealkv/mod.rs

```diff
@@ -0,0 +1,398 @@
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Check to see if transaction is writable
+		if !self.write {
+			return Err(Error::TxReadonly);
+		}
+		let ts = self.get_timestamp(ts_key).await?;
+		let mut k: Vec<u8> = prefix.into();
+		k.append(&mut ts.to_vec());
+		k.append(&mut suffix.into());
+		Ok(k)
+	}
+	/// Insert or update a key in the database
+	pub(crate) async fn set<K, V>(&mut self, key: K, val: V) -> Result<(), Error>
+	where
+		K: Into<Key>,
+		V: Into<Val>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Check to see if transaction is writable
+		if !self.write {
+			return Err(Error::TxReadonly);
+		}
+
+		// Set the key
+		self.inner.set(&key.into().as_slice(), &val.into())?;
+		// Return result
+		Ok(())
+	}
+
+	/// Insert a key if it doesn't exist in the database
+	pub(crate) async fn put<K, V>(
+		&mut self,
+		category: KeyCategory,
+		key: K,
+		val: V,
+	) -> Result<(), Error>
+	where
+		K: Into<Key>,
+		V: Into<Val>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+
+		// Check to see if transaction is writable
+		if !self.write {
+			return Err(Error::TxReadonly);
+		}
+
+		let key: Vec<u8> = key.into();
+		if self.exi(key.clone().as_slice()).await? {
+			return Err(Error::TxKeyAlreadyExistsCategory(category));
+		}
+
+		// Set the key
+		let key: &[u8] = &key[..];
+		self.inner.set(&key, &val.into())?;
+
+		// Return result
+		Ok(())
+	}
+	/// Insert a key if it doesn't exist in the database
+	pub(crate) async fn putc<K, V>(&mut self, key: K, val: V, chk: Option<V>) -> Result<(), Error>
+	where
+		K: Into<Key>,
+		V: Into<Val>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Check to see if transaction is writable
+		if !self.write {
+			return Err(Error::TxReadonly);
+		}
+
+		// Get the check
+		let chk = chk.map(Into::into);
+
+		// Set the key if not present
+		let key_slice = key.into();
+		let val_vec = val.into();
+
+		let res = self
+			.inner
+			.get(&key_slice.as_slice())
+			.map_err(|e| Error::Tx(format!("Unable to get kv from SurrealKV: {}", e)));
+
+		match (res, chk) {
+			(Ok(Some(v)), Some(w)) if v == w => self.inner.set(&key_slice.as_slice(), &val_vec),
+			(Ok(None), None) => self.inner.set(&key_slice.as_slice(), &val_vec),
+			(Err(e), _) => return Err(e),
+			_ => return Err(Error::TxConditionNotMet),
+		};
+
+		// Return result
+		Ok(())
+	}
+	/// Delete a key
+	pub(crate) async fn del<K>(&mut self, key: K) -> Result<(), Error>
+	where
+		K: Into<Key>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Check to see if transaction is writable
+		if !self.write {
+			return Err(Error::TxReadonly);
+		}
+		// Remove the key
+		let res = self.inner.delete(&key.into().as_slice())?;
```

Comment:
Same about return type. I would go over the return types and see if they match the wrapping functions return type. And then simply return the result instead of the `...?; Ok(_)` pattern

Reply:
you are right, fixing it